Detectors/TPC/qc/src/Tracks.cxx

```diff
@@ -0,0 +1,122 @@
+  mHist1D.emplace_back("hEta", "Pseudorapidity;eta", 400, -2., 2.);
+  mHist1D.emplace_back("hPhiAside", "Azimuthal angle, A side;phi", 628, -3.14, 3.14);
+  mHist1D.emplace_back("hPhiCside", "Azimuthal angle, C side;phi", 628, -3.14, 3.14);
+  mHist1D.emplace_back("hPt", "Transverse momentum;p_T", 200, 0., 10.);
+  mHist1D.emplace_back("hSign", "Sign of electric charge;charge sign", 3, -1.5, 1.5);
+
+  mHist2D.emplace_back("h2DNClustersEta", "Number of clusters vs. eta;eta;# TPC clusters", 400, -2., 2., 160, -0.5, 159.5);
+  mHist2D.emplace_back("h2DNClustersPhiAside", "Number of clusters vs. phi, A side ;phi;# TPC clusters", 628, -3.14, 3.14, 160, -0.5, 159.5);
+  mHist2D.emplace_back("h2DNClustersPhiCside", "Number of clusters vs. phi, C side ;phi;# TPC clusters", 628, -3.14, 3.14, 160, -0.5, 159.5);
+  mHist2D.emplace_back("h2DNClustersPt", "Number of clusters vs. p_T;p_T;# TPC clusters", 200, 0., 10., 160, -0.5, 159.5);
+  mHist2D.emplace_back("h2DEtaPhi", "Tracks in eta vs. phi;phi;eta", 628, -3.14, 3.14, 400, -2., 2.);
+  mHist2D.emplace_back("h2DEtaPhiNeg", "Negative tracks in eta vs. phi;phi;eta", 628, -3.14, 3.14, 400, -2., 2.);
+  mHist2D.emplace_back("h2DEtaPhiPos", "Positive tracks in eta vs. phi;phi;eta", 628, -3.14, 3.14, 400, -2., 2.);
+}
+
+//______________________________________________________________________________
+void Tracks::resetHistograms()
+{
+  for (auto& hist : mHist1D) {
+    hist.Reset();
+  }
+  for (auto& hist2 : mHist2D) {
+    hist2.Reset();
+  }
+}
+
+//______________________________________________________________________________
+bool Tracks::processTrack(const o2::tpc::TrackTPC& track)
+{
+  // ===| variables required for cutting and filling |===
+  const auto eta = track.getEta();
+  const auto phi = track.getPhi();
+  const auto pt = track.getPt();
+  const auto sign = track.getSign();
+  const auto nCls = track.getNClusterReferences();
+
+  // ===| fill one histogram before any cuts |===
+  mHist1D[0].Fill(nCls);
+
+  // ===| cuts |===
+  // hard coded cuts. Should be more configural in future
+  if (nCls < 20) {
+    return false;
+  }
+
+  // ===| 1D histogram filling |===
+  mHist1D[1].Fill(nCls);
+  mHist1D[2].Fill(eta);
+
+  if (eta > 0.) {
+    mHist1D[3].Fill(phi);
+  }
+  if (eta < 0.) {
+    mHist1D[4].Fill(phi);
+  }
+
+  mHist1D[5].Fill(pt);
+  mHist1D[6].Fill(sign);
+
+  // ===| 2D histogram filling |===
+  mHist2D[0].Fill(eta, nCls);
+
+  if (eta > 0.) {
+    mHist2D[1].Fill(phi, nCls);
+  }
+  if (eta < 0.) {
+    mHist2D[2].Fill(phi, nCls);
+  }
+
+  mHist2D[3].Fill(pt, nCls);
+  mHist2D[4].Fill(phi, eta);
+
+  if (sign < 0.) {
+    mHist2D[5].Fill(phi, eta);
+  }
+  if (sign > 0.) {
+    mHist2D[6].Fill(phi, eta);
+  }
+
+  return true;
+}
+
+//______________________________________________________________________________
+void Tracks::dumpToFile(std::string_view filename)
+{
+  auto f = std::unique_ptr<TFile>(TFile::Open(filename.data(), "recreate"));
```

Comment:
I think using `unique_ptr` only works because root includes it somewhere. To be more explicit I suggest to add
```c++
#include <memory>
```
above